Branch for pull request

ilyalopatin1994 opened the pull request:
В данном пул реквесте почти ничего нет.
Нужно смотреть ветку мастер.
Нужно было изначально создавать в другой ветке. Но я поздно об этом подумал.

Было сделано:
Компоненты: App.js Хедер. Страница с картой, Страница с логином, Страница с регистрацией.
Переход между страница осуществляется путем изменения стейта компонента App.js/
App.js - главный компонент. который рендерит остальные (пока нет роутера).

По заданию при нажатии ссылки "войти" на странице регистрации нужно переходить на страницу Карты. Я сделал переход на страницу логина, т.к. мне кажется это более логичным, а с точки зрения сложности это одно и то же, просто разные параметры.

shark132 committed:
Добавил страницу регистрации.
Сделал переходы

shark132 committed:
Revert "Добавил страницу регистрации."

This reverts commit ea6b5e37437a00f168d703c75a4bd94036d1ebc4.

shark132 committed:
Изменил фразу на странице с картой, чтобы создать пулл реквест в новой ветке

romradchenko commented:
Комментарии по коду:

1. Декомпозиция. Стоит разбивать на компоненты, создать директорию components в src и туда перекинуть все компоненты кроме App.js + также компоненты должны начинаться с большой буквы. 
2. Enum подобные структуры. Такой код лучше запихнуть в enum подобную структуру, то есть 
![image](https://user-images.githubusercontent.com/54706661/170017036-7838d621-c7c6-472f-ae53-58ddb785e097.png)
Вот пример реализации. Просто else if смотрится не очень, можно так. 
![image](https://user-images.githubusercontent.com/54706661/170017455-a39d1cf5-a75c-4c08-84c1-49a83978564b.png)

В Header.js тоже вроде можно лучше сделать. В общем посмотри сам.
3. Logo.js -> logo.svg. Не нужно делать logo в виде компонента. Лучше вынести в img/logo.svg и просто импортировать и использовать

romradchenko commented:
Дополнительно:
- Рекомендую commitzen установить https://tasks.loftschool.com/react_home/one.html#%D0%BF%D0%BE%D0%B4%D0%B3%D0%BE%D1%82%D0%BE%D0%B2%D0%BA%D0%B0-%D0%BA-%D0%B2%D1%8B%D0%BF%D0%BEn%D0%BD%D0%B5%D0%BD%D0%B8%D1%8E-%D0%B7%D0%B0%D0%B4%D0%B0%D0%BD%D0%B8%D1%8F

shark132 committed:
Внес изменения согласно замечаниям:
1) Переместил компоненты в папку components
2) Переименовал компонены (названия с большой буквы)
3) Сократил метод рендер в App.js, загнал компоненты в объект для удобного обращения
4) Подправил размер блока на странцие логина

shark132 committed:
Переименовал название css wrapper в login.css и registration.css, чтобы они не конфликтовали.

shark132 committed:
feat(module and file): first commitizen commit. Commitizen has been installed as a module dependency

shark132 committed:
refactor: config file

Some variables have been moved to the config.js file

shark132 committed:
fix(file directory): config.js moved to src/

shark132 committed:
Добавлен контекст авторизации.
Добавлена карта.

shark132 committed:
Добавлен контекст авторизации.
Добавлена карта.
Написаны тесты
